File: cmd/status.go
package cmd

import (
	"fmt"
	"os"

	"github.com/spf13/cobra"

	"github.com/pcr-developers/cli/internal/auth"
	"github.com/pcr-developers/cli/internal/projects"
	"github.com/pcr-developers/cli/internal/store"
)

var statusCmd = &cobra.Command{
	Use:   "status",
	Short: "Show auth, registered projects, and prompt bundle state",
	Run: func(cmd *cobra.Command, args []string) {
		const bold = "\x1b[1m"
		const dim = "\x1b[2m"
		const grn = "\x1b[32m"
		const yel = "\x1b[33m"
		const rst = "\x1b[0m"

		// ── Auth ──────────────────────────────────────────────────────────────
		a := auth.Load()
		if a != nil {
			fmt.Fprintf(os.Stderr, "%s✓%s Logged in (user: %s)\n", grn, rst, a.UserID)
		} else {
			fmt.Fprintln(os.Stderr, "Not logged in — run `pcr login`")
		}

		// ── Projects ──────────────────────────────────────────────────────────
		projs := projects.Load()
		fmt.Fprintln(os.Stderr)
		if len(projs) == 0 {
			fmt.Fprintln(os.Stderr, "No projects registered. Run `pcr init` in a project directory.")
		} else {
			fmt.Fprintf(os.Stderr, "%sProjects%s\n", bold, rst)
			for _, p := range projs {
				remote := ""
				if p.ProjectID != "" {
					remote = fmt.Sprintf("  %s[remote: %s]%s", dim, p.ProjectID, rst)
				}
				fmt.Fprintf(os.Stderr, "  %s%s\n  %s%s%s\n", p.Name, remote, dim, p.Path, rst)
			}
		}

		// ── Bundles ───────────────────────────────────────────────────────────
		unpushed, err := store.GetUnpushedCommits()
		if err == nil {
			fmt.Fprintln(os.Stderr)
			if len(unpushed) == 0 {
				fmt.Fprintf(os.Stderr, "%sBundles%s  none — everything pushed\n", bold, rst)
			} else {
				fmt.Fprintf(os.Stderr, "%sBundles%s\n", bold, rst)
				for _, b := range unpushed {
					full, _ := store.GetCommitWithItems(b.ID)
					count := 0
					if full != nil {
						count = len(full.Items)
					}
					if b.BundleStatus == "open" {
						fmt.Fprintf(os.Stderr, "  %s●%s  %s  %s(%d prompt%s)%s\n",
							yel, rst, b.Message, dim, count, plural(count), rst)
					} else {
						fmt.Fprintf(os.Stderr, "  %s✓%s  %s  %s(%d prompt%s — sealed, ready to push)%s\n",
							grn, rst, b.Message, dim, count, plural(count), rst)
					}
				}
			}
		}

		// ── Drafts ────────────────────────────────────────────────────────────
		drafts, err := store.GetDraftsByStatus(store.StatusDraft, nil, nil)
		if err == nil {
			fmt.Fprintln(os.Stderr)
			if len(drafts) == 0 {
				fmt.Fprintf(os.Stderr, "%sDrafts%s  none\n", bold, rst)
			} else {
				fmt.Fprintf(os.Stderr, "%sDrafts%s  %d unreviewed — run `pcr bundle` to create a prompt bundle\n", bold, rst, len(drafts))
			}
		}
	},
}

File: crates/pcr-core/src/commands/status.rs
//! `pcr status`. Mirrors `cli/cmd/status.go`.

use crate::agent::{self, OutputMode};
use crate::auth;
use crate::display::{self, Color};
use crate::exit::ExitCode;
use crate::projects;
use crate::store;
use crate::util::text::plural;

pub fn run(mode: OutputMode) -> ExitCode {
    if matches!(mode, OutputMode::Json) {
        return run_json();
    }
    if agent::is_tui_eligible(mode) {
        if let Err(e) = crate::tui::screens::status::run() {
            display::print_error("status", &e.to_string());
            return ExitCode::GenericError;
        }
        return ExitCode::Success;
    }
    run_plain()
}

fn run_plain() -> ExitCode {
    let a = auth::load();
    if let Some(a) = &a {
        display::eprintln(&format!(
            "{} Logged in (user: {})",
            display::cstr(Color::Green, "✓"),
            a.user_id
        ));
    } else {
        display::eprintln("Not logged in — run `pcr login`");
    }

    let projs = projects::load();
    display::eprintln("");
    if projs.is_empty() {
        display::eprintln("No projects registered. Run `pcr init` in a project directory.");
    } else {
        display::eprintln(&display::cstr(Color::Bold, "Projects"));
        for p in &projs {
            let remote = if p.project_id.is_empty() {
                String::new()
            } else {
                format!(
                    "  {}",
                    display::cstr(Color::Dim, &format!("[remote: {}]", p.project_id))
                )
            };
            display::eprintln(&format!("  {}{}", p.name, remote));
            display::eprintln(&format!("  {}", display::cstr(Color::Dim, &p.path)));
        }
    }

    if let Ok(unpushed) = store::get_unpushed_commits() {
        display::eprintln("");
        if unpushed.is_empty() {
            display::eprintln(&format!(
                "{}  none — everything pushed",
                display::cstr(Color::Bold, "Bundles")
            ));
        } else {
            display::eprintln(&display::cstr(Color::Bold, "Bundles"));
            for b in &unpushed {
                let full = store::get_commit_with_items(&b.id).ok().flatten();
                let count = full.map(|c| c.items.len()).unwrap_or(0);
                if b.bundle_status == "open" {
                    display::eprintln(&format!(
                        "  {}  {}  {}",
                        display::cstr(Color::Yellow, "●"),
                        b.message,
                        display::cstr(Color::Dim, &format!("({count} prompt{})", plural(count)))
                    ));
                } else {
                    display::eprintln(&format!(
                        "  {}  {}  {}",
                        display::cstr(Color::Green, "✓"),
                        b.message,
                        display::cstr(
                            Color::Dim,
                            &format!("({count} prompt{} — sealed, ready to push)", plural(count))
                        )
                    ));
                }
            }
        }
    }

    if let Ok(drafts) = store::get_drafts_by_status(store::DraftStatus::Draft, &[], &[]) {
        display::eprintln("");
        if drafts.is_empty() {
            display::eprintln(&format!("{}  none", display::cstr(Color::Bold, "Drafts")));
        } else {
            display::eprintln(&format!(
                "{}  {} unreviewed — run `pcr bundle` to create a prompt bundle",
                display::cstr(Color::Bold, "Drafts"),
                drafts.len()
            ));
        }
    }

    ExitCode::Success
}

fn run_json() -> ExitCode {
    let a = auth::load();
    let projs = projects::load();
    let unpushed = store::get_unpushed_commits().unwrap_or_default();
    let drafts =
        store::get_drafts_by_status(store::DraftStatus::Draft, &[], &[]).unwrap_or_default();
    let out = serde_json::json!({
        "logged_in": a.is_some(),
        "user_id": a.map(|a| a.user_id).unwrap_or_default(),
        "projects": projs,
        "unpushed_bundles": unpushed,
        "draft_count": drafts.len(),
    });
    println!("{}", serde_json::to_string_pretty(&out).unwrap_or_default());
    ExitCode::Success
}
